tracking-migration/src/main/java/io/customer/tracking/migration/MigrationAssistant.kt

```diff
@@ -0,0 +1,45 @@
+package io.customer.tracking.migration
+
+import android.content.Context
+import io.customer.tracking.migration.di.MigrationSDKComponent
+import io.customer.tracking.migration.queue.Queue
+import io.customer.tracking.migration.repository.preference.SitePreferenceRepository
+
+/**
+ * Class responsible for migrating the existing tracking data to the new data
+ * pipelines implementation.
+ */
+class MigrationAssistant(
+    context: Context,
+    private val migrationProcessor: MigrationProcessor,
+    migrationSiteId: String
+) {
+    private val migrationSDKComponent = MigrationSDKComponent(
+        applicationContext = context,
+        migrationProcessor = migrationProcessor,
+        migrationSiteId = migrationSiteId
+    )
+    private val sitePreferences: SitePreferenceRepository = migrationSDKComponent.sitePreferences
+    private val queue: Queue = migrationSDKComponent.queue
+
+    /**
+     * Starts the migration process by migrating the existing tracking data to the new
+     * data pipelines implementation using provided migration processor.
+     */
+    fun migrate() {
```

Comment:
can we call it `init` and avoid a method call? this class will only be initialized if `siteId` exists

Reply:
I avoid it as it reduces code readability. I added some comments and a helper method to improve readability with code in init block.